inspire_dojson/hep/rules/bdFFT.py

```diff
@@ -45,32 +46,49 @@ def _is_hidden(value):
         return 'HIDDEN' in [val.upper() for val in value] or None
 
     def _get_caption(value):
-        return ''.join(re.compile('^\d{5}\s+').split(value)[1:]) or None
+        return ''.join(re.compile('^\d{5}\s*').split(value)[1:]) or None
+
+    def _get_key(value):
+        fname = value.get('n', str(uuid.uuid4()))
+        extension = value.get('f', '')
+
+        if fname.endswith(extension):
+            return fname
+        return fname + extension
+
+    ordered_pattern = re.compile('^(\d{5})\s*')
+    figures_extensions = ['.png']
+    figures = self.get('figures', [])
 
     is_context = value.get('f', '').endswith('context')
+
     if is_context:
         return
 
-    type_ = value.get('t')
-    figures = self.get('figures', [])
-    document = None
-
-    if type_ == 'Plot':
-        figures.append({
-            'key': value.get('n') + value.get('f', ''),
-            'caption': _get_caption(value.get('d')),
-            'url': afs_url(value)
-        })
-    elif type_ == 'Main':
-        document = {
-            'description': value.get('d'),
-            'key': value.get('n') + value.get('f', ''),
-            'hidden': _is_hidden(force_list(value.get('o'))),
-            'url': afs_url(value)
-        }
+    format_ = value.get('f', '')
+    ordnum_match = ordered_pattern.match(value.get('d'))
+    if format_ in figures_extensions:
+        if ordnum_match:
+            figures.append({
+                'key': _get_key(value),
+                'caption': _get_caption(value.get('d')),
+                'url': afs_url(value),
+                'order': ordnum_match.group(1)
+            })
+        else:
+            figures.append({
+                'key': _get_key(value),
+                'caption': _get_caption(value.get('d')),
+                'url': afs_url(value)
+            })
 
     self['figures'] = figures
-    return document
+    return {
```

Comment:
this looks wrong: if it's a figure, it also gets executed. You should assert in the tests that when you populate `figures`, `'documents' not in record` and vice-versa.